dbms/src/Storages/StorageMaterializedView.cpp

```diff
@@ -265,6 +270,51 @@ void StorageMaterializedView::mutate(const MutationCommands & commands, const Co
     getTargetTable()->mutate(commands, context);
 }
 
+static void executeRenameQuery(Context & global_context, const String & database_name, const String & table_original_name, const String & new_table_name)
+{
+    if (global_context.tryGetTable(database_name, table_original_name))
+    {
+            auto rename = std::make_shared<ASTRenameQuery>();
+
+            ASTRenameQuery::Table from;
+            from.database = database_name;
+            from.table = table_original_name;
+
+            ASTRenameQuery::Table to;
+            to.database = database_name;
+            to.table = new_table_name;
+
+            ASTRenameQuery::Element elem;
+            elem.from = from;
+            elem.to = to;
+
+            rename->elements.emplace_back(elem);
+
+            InterpreterRenameQuery(rename, global_context).execute();
+    }
+}
+
+
+void StorageMaterializedView::rename(const String & /*new_path_to_db*/, const String & /*new_database_name*/, const String & new_table_name)
+{
+    if (has_inner_table && tryGetTargetTable())
+    {
+        String new_target_table_name = generateInnerTableName(new_table_name);
+        executeRenameQuery(global_context, target_database_name, target_table_name, new_target_table_name);
+        target_table_name = new_target_table_name;
+    }
+    
+    global_context.removeDependency(
```

Comment:
Maybe should use `global_context.getLock()` to get the context lock first ?

Reply:
`Context::getLock()` is call on both `removeDependency` and `addDependency` function member. Or you mean to have the same lock for both operation ?

Reply:
Yes, because renaming is atomic, do we guarantee that materialized views are writable during renaming?

Reply:
Ok thanks got it now :)

I applied the change about it, better to add a lock for not letting people write on the materialized views during renaming.